src/ncp-spinel/SpinelNCPInstance.cpp

```diff
@@ -2556,93 +2539,183 @@ SpinelNCPInstance::handle_ncp_spinel_callback(unsigned int command, const uint8_
 	process_event(EVENT_NCP(command), cmd_data_ptr[0], cmd_data_ptr, cmd_data_len);
 }
 
-void
-SpinelNCPInstance::address_was_added(const struct in6_addr& addr, int prefix_len)
+bool
+SpinelNCPInstance::should_filter_address(const struct in6_addr &addr, uint8_t prefix_len)
```

Comment:
The full RLOC IPv6 addresses are a thread implementation detail. Other types of networks wouldn't have them, and including them here requires a special case for thread networks. I'd rather that special case be in the NCP at some point in the near future.